LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2LWAbilitiesModTemplate.uc

```diff
@@ -559,7 +568,86 @@ static function AddImmuneConditionToFlamethrower(X2AbilityTemplate Template)
 
 	
 
+static function	UpdatePurifierFlamethrower(X2AbilityTemplate Template)
+{
+	local X2AbilityTarget_Cursor				CursorTarget;
+	local X2AbilityMultiTarget_Cone_LWFlamethrower	ConeMultiTarget;
+	local X2Effect_ApplyFireToWorld_Limited		FireToWorldEffect;
+	local X2AbilityToHitCalc_StandardAim		StandardAim;
+	local array<name>                       SkipExclusions;
+
+
+	StandardAim = new class'X2AbilityToHitCalc_StandardAim';
+	StandardAim.bAllowCrit = false;
+	StandardAim.bGuaranteedHit = true;
+	Template.AbilityToHitCalc = StandardAim;
+
+	SkipExclusions.AddItem(class'X2AbilityTemplateManager'.default.DisorientedName);
+	Template.AddShooterEffectExclusions(SkipExclusions);
+
+	Template.TargetingMethod = class'X2TargetingMethod_Cone_Flamethrower_LW';
+
+	ConeMultiTarget = new class'X2AbilityMultiTarget_Cone_LWFlamethrower';
+	ConeMultiTarget.bUseWeaponRadius = true;
 
+	ConeMultiTarget.ConeEndDiameter = class'X2Ability_AdvPurifier'.default.ADVPURIFIER_FLAMETHROWER_TILE_WIDTH * class'XComWorldData'.const.WORLD_StepSize;
+	ConeMultiTarget.ConeLength = class'X2Ability_AdvPurifier'.default.ADVPURIFIER_FLAMETHROWER_TILE_LENGTH * class'XComWorldData'.const.WORLD_StepSize;
+	ConeMultiTarget.AddConeSizeMultiplier('Incinerator', class'X2Ability_LW_TechnicalAbilitySet'.default.INCINERATOR_RANGE_MULTIPLIER, class'X2Ability_LW_TechnicalAbilitySet'.default.INCINERATOR_RADIUS_MULTIPLIER);
+	// Next line used for vanilla targeting
+	// ConeMultiTarget.AddConeSizeMultiplier('Incinerator', default.INCINERATOR_CONEEND_DIAMETER_MODIFIER, default.INCINERATOR_CONELENGTH_MODIFIER);
+	ConeMultiTarget.bIgnoreBlockingCover = true;
+	Template.AbilityMultiTargetStyle = ConeMultiTarget;
+
+	Template.bCheckCollision = true;
+	Template.bAffectNeighboringTiles = true;
+	Template.bFragileDamageOnly = true;
+
+	// For vanilla targeting
+	// Template.ActionFireClass = class'X2Action_Fire_Flamethrower';
+	Template.PostActivationEvents.AddItem('FlamethrowerActivated');
+	Template.ActionFireClass = class'X2Action_Fire_Flamethrower_LW';
+
+	Template.BuildVisualizationFn = class'X2Ability_LW_TechnicalAbilitySet'.static.LWFlamethrower_BuildVisualization;
```

Comment:
If this is identical to the flamethrower setup code for Technical flamethrower, apart from the size, it should really be factored into a static helper function with the cone length and diameter as arguments. Otherwise folks need to remember to change things in two different places.

Reply:
You can also add the disorient exclusion to the template after it's returned from the helper function.

Reply:
It is not identical, the template effects are different. I think these two should actually be kepts seperated as they shouldn't exactly be standarized.

Reply:
Fair enough